app/src/main/java/com/bc/gordiansigner/service/AccountMapService.kt

```diff
@@ -0,0 +1,49 @@
+package com.bc.gordiansigner.service
+
+import com.bc.gordiansigner.helper.ext.newGsonInstance
+import com.bc.gordiansigner.model.AccountMap
+import com.bc.gordiansigner.model.Descriptor
+import com.bc.gordiansigner.model.HDKey
+import com.bc.gordiansigner.service.storage.FileStorageApi
+import com.bc.gordiansigner.service.storage.SharedPrefApi
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import javax.inject.Inject
+
+class AccountMapService @Inject constructor(
+    sharedPrefApi: SharedPrefApi,
+    fileStorageApi: FileStorageApi
+) : BaseService(sharedPrefApi, fileStorageApi) {
+
+    fun saveAccountMap(accountMap: AccountMap): Single<AccountMap> {
+        //TODO: save account map, implement later
+
+        return Single.just(accountMap)
+    }
+
+    fun getAccountMapInfo(accountMapString: String): Single<Pair<AccountMap, Descriptor>> =
+        Single.fromCallable {
+            val accountMap = newGsonInstance().fromJson(accountMapString, AccountMap::class.java)
+            val descriptor = Descriptor.fromString(accountMap.descriptor)
+
+            Pair(accountMap, descriptor)
+        }.subscribeOn(Schedulers.io())
+
+    fun fillPartialAccountMap(
+        accountMap: AccountMap,
+        descriptor: Descriptor,
+        hdKey: HDKey
+    ): Single<String> {
+        return Single.fromCallable {
+            if (!descriptor.isCompleted()) {
+                descriptor.updatePartialAccountMapFromKey(hdKey)
+            }
+            AccountMap(descriptor.toString(), accountMap.blockheight, accountMap.label)
```

Comment:
I think there was mistake in this rx implementation here. `AccountMap` observable  would not be triggered since it has not been subscribe anymore.

Reply:
I made this account map returns to the Single.fromCallalbe, did you see anything's incorrect here?

Reply:
OK. That's my mistake in code review.